packages/components/src/internal/components/samples/SampleAliquotsGridPanel.tsx

```diff
@@ -0,0 +1,157 @@
+            rootLsid,
+            getOmittedSampleTypeColumns(user, inventoryCols)
+        );
+        // don't need to load the data here because that is done by default in the GridPanel.
+        actions.addModel(queryConfig, false);
+    }
+
+    getQueryModel(): QueryModel {
+        return Object.values(this.props.queryModels)[0];
+    }
+
+    afterAction = () => {
+        const { actions, schemaQuery, onSampleChangeInvalidate } = this.props;
+        this.resetState();
+        onSampleChangeInvalidate(schemaQuery);
+        const model = this.getQueryModel();
+        actions.loadModel(model.id, true);
+    };
+
+    hideConfirm = () => {
+        this.setState(() => ({ showConfirmDelete: false }));
+    };
+
+    onDelete = () => {
+        if (this.hasSelection()) {
+            this.setState(() => ({ showConfirmDelete: true }));
+        }
+    };
+
+    resetState = () => {
+        if (this.hasSelection()) {
+            this.setState(() => ({ showConfirmDelete: false }));
+        }
+    };
+
+    hasSelection() {
+        return this.getQueryModel().selections?.size > 0;
+    }
+
+    renderButtons = () => {
+        const queryModel = this.getQueryModel();
+        const Node = this.props.storageButton;
+        return (
+            <div className="btn-group">
+                <RequiresPermission perms={PermissionTypes.Delete}>
+                    <ManageDropdownButton id="samplealiquotlisting">
+                        <SelectionMenuItem
+                            id="sample-aliquot-delete-menu-item"
+                            text="Delete Aliquots"
+                            onClick={() => this.onDelete()}
+                            queryModel={queryModel}
+                            nounPlural="aliquots"
+                        />
+                    </ManageDropdownButton>
+
+                    {Node && (
+                        <Node user={this.props.user} queryModel={queryModel} afterStorageUpdate={this.afterAction} />
+                    )}
+                </RequiresPermission>
+            </div>
+        );
+    };
+
+    renderDeleteModal() {
+        const model = this.getQueryModel();
+
+        if (!this.state.showConfirmDelete) return null;
+
+        return (
+            <EntityDeleteModal
+                queryModel={model}
+                useSelected={true}
+                afterDelete={this.afterAction}
+                onCancel={this.resetState}
+                entityDataType={SampleTypeDataType}
+                verb="deleted and removed from storage"
+            />
+        );
+    }
+
+    render() {
+        const { actions } = this.props;
+
+        const model = this.getQueryModel();
+
+        if (!model) return <LoadingSpinner />;
+
+        return (
+            <>
+                {this.state.showConfirmDelete && this.renderDeleteModal()}
+                <GridPanel
+                    actions={actions}
+                    ButtonsComponent={() => this.renderButtons()}
```

Comment:
For `ButtonsComponent` and `buttonsComponentProps` the GridPanel expects to be able to instantiate an instance of the component handed to `ButtonsComponent` with the props provided in `buttonsComponentProps`. Additionally, `GridPanel` will pass in the associated `actions` and `model` for the component to use via props.

For clarity of use I would recommend creating a separate component type and then pass the configuration through props. Here is an example:

```tsx
interface StorageButtonsComponentProps {
    afterStorageUpdate: () => void;
    queryModel: QueryModel;
    user: User;
}

type StorageButton = ComponentType<StorageButtonsComponentProps>;

interface AliquotGridButtonsProps {
    afterAction: () => void;
    onDelete: () => void;
    StorageButtonsComponent?: StorageButton;
    user: User;
}

const AliquotGridButtons: FC<AliquotGridButtonsProps & RequiresModelAndActions> = props => {
    const { afterAction, model, onDelete, StorageButtonsComponent, user } = props;

    return (
        <div className="btn-group">
            <RequiresPermission perms={PermissionTypes.Delete}>
                <ManageDropdownButton id="samplealiquotlisting">
                    <SelectionMenuItem
                        id="sample-aliquot-delete-menu-item"
                        text="Delete Aliquots"
                        onClick={onDelete}
                        queryModel={model}
                        nounPlural="aliquots"
                    />
                </ManageDropdownButton>

                {StorageButtonsComponent && (
                    <StorageButtonsComponent afterStorageUpdate={afterAction} queryModel={model} user={user} />
                )}
            </RequiresPermission>
        </div>
    );
};
```

The `render()` method would look like:

```tsx
render() {
    const { actions, queryModels, storageButton, user } = this.props;
    const { model } = queryModels;

    return (
        <>
            <GridPanel
                actions={actions}
                ButtonsComponent={AliquotGridButtons}
                buttonsComponentProps={{
                    afterAction: this.afterAction,
                    onDelete: this.onDelete,
                    StorageButtonsComponent: storageButton,
                    user,
                }}
                loadOnMount={false}
                model={model}
                showViewMenu={false}
            />

            {this.state.showConfirmDelete && (
                <EntityDeleteModal
                    afterDelete={this.afterAction}
                    entityDataType={SampleTypeDataType}
                    onCancel={this.resetState}
                    queryModel={model}
                    useSelected
                    verb="deleted and removed from storage"
                />
            )}
        </>
    );
}
```

Reply:
Done (in patch)